notification/source/c/src/kaa_demo.c

```diff
@@ -65,35 +66,26 @@ static int read_topic_id(void)
     return atoi(input);
 }
 
-static kaa_topic_t *find_topic(uint64_t topic_id)
+static bool find_topic_predicate(void *data, void *context)
 {
-    if (!topic_list || !kaa_list_get_size(topic_list)) {
-        return NULL;
-    }
-    kaa_list_node_t *it = kaa_list_begin(topic_list);
-    while (it) {
-        kaa_topic_t *t = (kaa_topic_t *)kaa_list_get_data(it);
-        if (t->id == topic_id) {
-            return t;
-        }
-        it = kaa_list_next(it);
-    }
-    return NULL;
+    kaa_topic_t *t = (kaa_topic_t *)data;
+    return t->id == *(int64_t*)context;
 }
 
 static void on_notification(void *context, uint64_t *topic_id, kaa_notification_t *notification)
 {
     (void)context;
     if (notification->alert_message) {
         kaa_string_t *message = (kaa_string_t *)notification->alert_message;
-        demo_printf("Notification for topic id '%lu' received\r\n", *topic_id);
+        demo_printf("Notification for topic id '%li' received\r\n", *topic_id);
```

Comment:
`%li` is for `long int`. `sizeof(long int)` for x86_64 is 4. Have you seen my previous comment?